api/v1alpha1/sandbox_types.go

```diff
@@ -58,6 +58,17 @@ type PodTemplate struct {
 	ObjectMeta PodMetadata `json:"metadata" protobuf:"bytes,3,opt,name=metadata"`
 }
 
+type PersistentVolumeClaimTemplate struct {
+	// Metadata is the Pod's metadata. Only labels and annotations are used.
+	// +kubebuilder:pruning:PreserveUnknownFields
```

Comment:
Since `metav1.ObjectMeta` is defined here, I would expect it to generate a full structural schema for all of its fields. This should make `PreserveUnknownFields` unnecessary. 

Is there a specific reason this marker is needed? It's generally a best practice to rely on the structural schema for pruning and avoid `preserveUnknownFields` when possible.

Reply:
Doesnt work. I think it is something to do with how core types are annotated and controller-gen is implemented. 

```
% kubectl apply -f examples/sandbox.yaml 
Error from server (BadRequest): error when creating "examples/sandbox.yaml": Sandbox in version "v1alpha1" cannot be handled as a Sandbox: strict decoding error: unknown field "spec.volumeClaimTemplates[0].metadata.name"
```

refer: #39 and https://github.com/kubernetes-sigs/kubebuilder/discussions/2726

Reply:
I am not sure https://github.com/kubernetes-sigs/kubebuilder/discussions/2726 is good precedent.

I agree with Janet that types should work here, I would expect to see metadata.name in the CRD.